src/commands/inhouse.js

```diff
@@ -0,0 +1,271 @@
+								let team2Names = [];
+								serverInhouses[message.guild.name].currentMatch.team1.forEach(element => {
+									team1Names.push(element.summonerName);
+								});
+								serverInhouses[message.guild.name].currentMatch.team2.forEach(element => {
+									team2Names.push(element.summonerName);
+								});
+								if (serverInhouses[message.guild.name].currentMatch !== null) {
+									let desc = ((serverInhouses[message.guild.name].type === '4fun')?'Sorted randomly':'Sorted by Soloqueue Rank');
+									message.channel.send('',
+										new Discord.RichEmbed({title: 'Lineups: ', description: desc})
+											.addField('Team 1:', team1Names.join('\n'), true)
+											.addField('Team 2:', team2Names.join('\n')), true);
+								}
+								logger.info('Inhouse full for server ' + message.guild.name + ', created teams...');
+							}
+							if (joinReturnCode === 1) {	
+								let joinedNames = [];
+								serverInhouses[message.guild.name].players.forEach(element => {
+									joinedNames.push(element.summonerName);
+								});
+
+								if (!(message.guild.name in serverInhouseMessageIDs)) {
+									message.delete();
+									message.channel.send('',
+										new Discord.RichEmbed({title: 'Current Players:'})
+											.addField(joinedNames.join(', '), false)
+									).then(respone => {
+										serverInhouseMessageIDs[message.guild.name] = respone.id;
+									});
+								} else {
+									message.delete();
+									message.channel.fetchMessage(serverInhouseMessageIDs[message.guild.name])
+										.then(message => message.edit('',
+											new Discord.RichEmbed({title: 'Current Players:'})
+												.addField(joinedNames.join(', '), false)));
+								}
+							}
+						});
+					} else {
+						message.channel.send('Please include your League of Legends summoner name like so: !f inhouse join <summoner name>');
+					}
+					break;
+				} 
+			} catch (err) {
+				logger.error(err);
+			}
+		}
+	} };
+
+class Inhouse {
+	constructor(_type) {
+		this.players = [];
+		this.currentMatch = null;
+		this.type = _type;
+	}
+
+	addPlayerToInhouse(newPLayerDiscordName, newPlayerSummonerName) {
+		return new Promise((resolve) =>{
+			new Player(newPLayerDiscordName, newPlayerSummonerName).then(response => {
+				let newPlayer = response;
+				if (newPlayer.summonerID === -1) {
+					resolve(-1);
+				}             
+				this.players.push(newPlayer);
+				if (this.players.length === 10 && this.currentMatch === null) {
+					this.CreateMatch();
+					resolve(0);
+				}
+    
+				resolve(1);
+			});
+		});   
+	}
+
+	CreateMatch() {
+		this.currentMatch = new Match(this.players, this.type);
+	}
+}
+
+class Player {
+	constructor(playerDiscordName, playerSummonerName) {
+		return new Promise((resolve) => {
+			this.discordName = playerDiscordName;
+			this.summonerName = playerSummonerName;
+			this.summonerID;
+			this.rankWeight;
+    
+			this.getSummonerId(this.summonerName).then(response => {
```

Comment:
this is callback hell btw.

make sure you do a .catch() after each of your .thens so that we handle any failed promises. can just do `logger.error` inside of those catches.